integration/nydus/nydus_tests.sh

```diff
@@ -0,0 +1,192 @@
+#!/bin/bash
+#
+# Copyright (c) 2022 Ant Group
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This will test the nydus feature is working properly
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o errtrace
+
+dir_path=$(dirname "$0")
+source "${dir_path}/../../lib/common.bash"
+source "${dir_path}/../../.ci/lib.sh"
+#cidir="${dir_path}/../../.ci"
+#echo "cidir $cidir"
+KATA_HYPERVISOR="${KATA_HYPERVISOR:-qemu}"
+
+if [ "$KATA_HYPERVISOR" != "qemu" ]; then
+	echo "Skip nydus test for $KATA_HYPERVISOR, it only works for QEMU now."
+	exit 0
+fi
+
+arch="$(uname -m)"
+if [ "$arch" != "x86_64" ]; then
+	echo "Skip nydus test for $arch, it only works for x86_64 now."
+	exit 0
+fi
+
+function setup_nydus() {
+	# install nydus
+	nydus_tarball_url=$(get_version "externals.nydus.url")
+	nydus_version=$(get_version "externals.nydus.version")
+	tarball_url="${nydus_tarball_url}/releases/download/${nydus_version}/nydus-static-${nydus_version}-x86_64.tgz"
+	echo "Download tarball from ${tarball_url}"
+	curl -Ls "$tarball_url" | sudo tar xfz - -C /usr/local/bin --strip-components=1
+
+	# FIXME install nydus-snapshotter from contanierd
+	# Now nydus-snapshotter can be installed from nydus(image-service)
+
+	# Config nydus snapshotter
+	cp "$dir_path/nydusd-config.json" /etc/
+
+	# start nydus-snapshotter
+	nohup /usr/local/bin/containerd-nydus-grpc \
+	    --config-path /etc/nydusd-config.json \
+	    --shared-daemon \
+	    --log-level debug \
+	    --root /var/lib/containerd/io.containerd.snapshotter.v1.nydus \
+	    --cache-dir /var/lib/nydus/cache \
+	    --nydusd-path /usr/local/bin/nydusd-fusedev \
+	    --nydusimg-path /usr/local/bin/nydus-image \
+	    --disable-cache-manager true \
+	    --enable-nydus-overlayfs true  \
+	    --log-to-stdout > /dev/null 2>&1 &
+}
+
+need_restore_kata_config=false
+kata_config_backup="/tmp/kata-configuration.toml"
+SYSCONFIG_FILE="/etc/kata-containers/configuration.toml"
+DEFAULT_CONFIG_FILE="/usr/share/defaults/kata-containers/configuration-qemu.toml"
```

Comment:
Could you put these variables at the top of the file? Same comment for the block below.

Reply:
moved